services/app-api/src/postgres/contractAndRates/unlockContract.test.ts

```diff
@@ -0,0 +1,346 @@
+// // For use in TESTS only. Throws a returned error
+import { sharedTestPrismaClient } from '../../testHelpers/storeHelpers'
+import { v4 as uuidv4 } from 'uuid'
+import { insertDraftContract } from './insertContract'
+import { unlockContract } from './unlockContract'
+import { insertDraftRate } from './insertRate'
+import { unlockRate } from './unlockRate'
+import { findDraftContract } from './findDraftContract'
+import { submitRateRevision } from './submitRateRevision'
+import { updateContractDraft } from './updateContractDraft'
+import { updateRateDraft } from './updateRateDraft'
+import { submitContract } from './submitContract'
+import { findContract } from './findContract'
+import { must } from '../../testHelpers'
+
+describe('unlockContract', () => {
+    it('Unlocks a rate without breaking connected draft contract', async () => {
+        const client = await sharedTestPrismaClient()
+
+        const stateUser = await client.user.create({
+            data: {
+                id: uuidv4(),
+                givenName: 'Aang',
+                familyName: 'Avatar',
+                email: 'aang@example.com',
+                role: 'STATE_USER',
+                stateCode: 'NM',
+            },
+        })
+
+        const cmsUser = await client.user.create({
+            data: {
+                id: uuidv4(),
+                givenName: 'Zuko',
+                familyName: 'Hotman',
+                email: 'zuko@example.com',
+                role: 'CMS_USER',
+            },
+        })
+
+        //Creat a draft contract and draft rate
+        const contract = must(await insertDraftContract(client, 'Contract 1.0'))
+        const rate = must(await insertDraftRate(client, 'Rate 1.0'))
+
+        // Submit Rate A
+        const submittedRateRev = must(
+            await submitRateRevision(
+                client,
+                rate.id,
+                stateUser.id,
+                'Rate A 1.0 submit'
+            )
+        )
+
+        // Connect draft contract to submitted rate
+        must(
+            await updateContractDraft(client, contract.id, 'Connecting rate', [
+                rate.id,
+            ])
+        )
+
+        const draftContract = must(await findDraftContract(client, contract.id))
+
+        if (draftContract === undefined) {
+            if (draftContract === undefined) {
+                throw Error('Contract data was undefined')
+            }
+        }
+
+        // Rate revision should be connected to contract
+        expect(draftContract.rateRevisions[0].id).toEqual(submittedRateRev.id)
```

Comment:
I _think_ this shouldn't work? draftContract is only going to have draftRates. Might not be returning the right thing, here.

Reply:
It looks like its working?

![Screenshot 2023-06-28 at 11 52 21 AM](https://github.com/Enterprise-CMCS/managed-care-review/assets/98117700/6cdd1665-427f-439c-ae1d-e0f1345dd777)

Submitted Rate
```json
{
   "id":"ca6e933f-623c-4da0-b5df-3879be6ae6c1",
   "revisions":[
      {
         "id":"25cc195b-d389-4931-a07a-f68de2974eb0",
         "submitInfo":{
            "updatedAt":"2023-06-28T15:51:16.018Z",
            "updatedBy":"4b5a3ab3-ae41-4d8b-b0b1-910925bc1b03",
            "updatedReason":"Rate A 1.0 submit"
         },
         "revisionFormData":"Rate 1.0",
         "contractRevisions":[
            
         ]
      }
   ]
}
```

Draft Contract
```json
{
   "id":"6742d8b4-d3a1-4874-a431-9ebb76557c48",
   "contractFormData":"Connecting rate",
   "rateRevisions":[
      {
         "id":"25cc195b-d389-4931-a07a-f68de2974eb0",
         "revisionFormData":"Rate 1.0"
      }
   ]
}
```

Reply:
Interesting, I'll take a look at this myself after it merges and think about it